solr/core/src/java/org/apache/solr/request/macro/MacroExpander.java

```diff
@@ -154,14 +155,14 @@ else if (idx < 0) {
       }
 
       if (matchedStart > 0) {
-        sb.append(val.substring(start, matchedStart));
+        sb.append(val, start, matchedStart);
       }
 
       // update "start" to be at the end of ${...}
-      start = rbrace + 1;
+      idx = start = rbrace + 1;
 
-      // String inbetween = val.substring(idx, rbrace);
-      StrParser parser = new StrParser(val, idx, rbrace);
+      // String in-between = val.substring(idx, rbrace);
```

Comment:
How about removing rather than amending the `// String inbetween = val.substring(idx, rbrace);` commented out code line?